### DISCUSSION TITLE
Multiple problems following the chirpy-starter upgrade guide

### DISCUSSION THREAD

**Discussion by seanthegeek** [cotes2020/jekyll-theme-chirpy#2005] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

I am preparing for future upgrades to Chirpy by following the upgrade guide. I'm getting the following error when trying to run `git fetch chirpy --tags`

### Operations you have already tried

I have followed the upgrade instructions from [the wiki](https://github.com/cotes2020/jekyll-theme-chirpy/wiki/Upgrade-Guide#upgrade-to-the-latest-version)

> git remote add chirpy git@github.com:cotes2020/chirpy-starter.git
> git remote -v

>chirpy  git@github.com:cotes2020/chirpy-starter.git (fetch)
> chirpy  git@github.com:cotes2020/chirpy-starter.git (push)
> origin  https://github.com/seanthegeek/seanthegeek.github.io.git (fetch)
> origin  https://github.com/seanthegeek/seanthegeek.github.io.git (push)

> git fetch chirpy --tags

>git@github.com: Permission denied (publickey).
>fatal: Could not read from remote repository.
>
>Please make sure you have the correct access rights
>and the repository exists.

I'm not sure why I'm getting a permission denied error for a public repository.

### Anything else?

 I am using OAuth-based authentication using GitHub's [gh utility](https://cli.github.com/).

**Comment by cotes2020** [comment]
> git remote -v
>
> chirpy [git@github.com](mailto:git@github.com):cotes2020/chirpy-starter.git (fetch)
> chirpy [git@github.com](mailto:git@github.com):cotes2020/chirpy-starter.git (push)
> origin https://github.com/seanthegeek/seanthegeek.github.io.git (fetch)
> origin https://github.com/seanthegeek/seanthegeek.github.io.git (push)

1. [Adding a new SSH key to your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
2. As an alternative, change the Chirpy upstream URL to <https://github.com/cotes2020/chirpy-starter.git>

**Reply by seanthegeek** [reply]
I already have a SSH key on my account. Apparently using `git@github.com:cotes2020/chirpy-starter.git` requires maintainer access to the repo.

Using the HTTPS URL works. Let's update the wiki

**Reply by seanthegeek** [reply]
The `git merge` command also needs `--allow-unrelated-histories` added to it.

**Reply by seanthegeek** [reply]
Another issue, even after resolving all other conflicts, git will not allow a pull.

```text
git -C assets/lib pull

error: Pulling is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
```

The only merge conflict is `assets/lib`.

```text
git status

Your branch is up to date with 'origin/main'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        new file:   .devcontainer/devcontainer.json
        new file:   .devcontainer/post-create.sh
        modified:   .github/workflows/pages-deploy.yml
        modified:   .gitignore
        new file:   .vscode/extensions.json
        modified:   .vscode/settings.json
        new file:   .vscode/tasks.json
        modified:   Gemfile
        modified:   README.md
        modified:   _config.yml
        modified:   _data/contact.yml
        modified:   _data/share.yml
        new file:   _posts/.placeholder
        new file:   tools/run.sh
        new file:   tools/test.sh

git commit -m "Chirpy to 7.1.1"
U       assets/lib
error: Committing is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
```

**Reply by seanthegeek** [reply, marked as answer]
I was able to upgrade successfully using the following steps.

1. If this is your first time upgrading, initialize the `assets/lib` submodule and add the remote
> git submodule update --init --recursive
> git remote add chirpy https://github.com/cotes2020/chirpy-starter.git
2. List the remote tags
> git fetch chirpy --tags
3. Merge with a remote tag
> git merge --squash --allow-unrelated-histories vx.y.z
4. Update the submodule
> git submodule update --recursive
5. Make changes as needed to address merge conflicts
6. Update gems
> bundle update
7. Test changes
> ./tools/test.sh
8. Add all changes, commit, and push
> git add . && git commit -m "Upgrade to Chirpy X.Y.Z" && git push

@cotes2020 Please consider updating the wiki with these steps.

**Reply by mattpopovich** [reply]
I also had to manually populate the [`.gitmodules`](https://github.com/cotes2020/chirpy-starter/blob/main/.gitmodules) file for some reason.
```
[submodule "assets/lib"]
	path = assets/lib
	url = https://github.com/cotes2020/chirpy-static-assets.git
```
